sonar-orchestrator/src/main/java/com/sonar/orchestrator/http/HttpClient.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Orchestrator
+ * Copyright (C) 2011-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.orchestrator.http;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.concurrent.TimeUnit;
+import okhttp3.Authenticator;
+import okhttp3.Credentials;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.Route;
+
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+
+public class HttpClient {
+
+  private final OkHttpClient okClient;
+
+  private HttpClient(OkHttpClient okClient) {
+    this.okClient = okClient;
+  }
+
+  public HttpCall newCall(HttpUrl url) {
+    return new HttpCall(okClient, url);
+  }
+
+  @VisibleForTesting
+  OkHttpClient getUnderlying() {
+    return okClient;
+  }
+
+  public static class Builder {
+    private SystemProperties system = SystemProperties.INSTANCE;
+
+    public Builder() {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1186)